File: sdklibrary/src/main/java/com/loner/android/sdk/webservice/network/helper/NetworkErrorInformation.java
package com.loner.android.sdk.webservice.network.helper;

import org.json.JSONException;
import org.json.JSONObject;


/**
 * This class stores all the information's related to the Errors that can occur.
 *
 * @author Swapnil Sonar
 * @version 1.0.0
 * @Date 29/12/2015
 */
public class NetworkErrorInformation extends BaseJSONParser {

    private final String KEY_STATUS_CODE = "statusCode";
    private final String KEY_ERROR = "error";
    private final String KEY_MESSAGE = "message";
    private final String KEY_DETAILS = "details";
    private final String KEY_DETAILS_CODE = "code";
    private final String KEY_DETAILS_MESSAGE = "message";

    private int statusCode;
    private String error;
    private String message;
    private int detailCode;
    private String detailMessage;

    public NetworkErrorInformation() {
        //Empty Constructor
    }

    public NetworkErrorInformation(byte[] responseData) {
        String responseString = new String(responseData);
        JSONObject jsonRegister;

        try {
            jsonRegister = new JSONObject(responseString);

            statusCode = Integer.parseInt(optInt(jsonRegister, KEY_STATUS_CODE));
            error = optString(jsonRegister, KEY_ERROR);
            message = optString(jsonRegister, KEY_MESSAGE);

            JSONObject detailObject = optJSONObject(jsonRegister, KEY_DETAILS);
            if (detailObject != null) {

                try {
                    detailCode = Integer.parseInt(optInt(detailObject, KEY_DETAILS_CODE));
                } catch (Exception e) {
                    e.printStackTrace();
                    detailCode = 0;
                }

                try {
                    detailMessage = optString(detailObject, KEY_DETAILS_MESSAGE);
                } catch (Exception e) {
                    e.printStackTrace();
                    detailMessage = "Unspecified error";
                }
            }

        } catch (JSONException e) {
            e.printStackTrace();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public int getStatusCode() {
        return statusCode;
    }

    public void setStatusCode(int statusCode) {
        this.statusCode = statusCode;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getDetailCode() {
        return detailCode;
    }

    public void setDetailCode(int detailCode) {
        this.detailCode = detailCode;
    }

    public String getDetailMessage() {
        return detailMessage;
    }

    public void setDetailMessage(String detailMessage) {
        this.detailMessage = detailMessage;
    }

    @Override
    public String toString() {
        return "NetworkErrorInformation{" +
                "statusCode=" + statusCode +
                ", error='" + error + '\'' +
                ", message='" + message + '\'' +
                ", detailCode=" + detailCode +
                ", detailMessage='" + detailMessage + '\'' +
                '}';
    }
}

File: LonerSdkLibrary/src/main/java/com/loner/android/sdk/webservice/network/helper/NetworkErrorInformation.kt
package com.loner.android.sdk.webservice.network.helper

import org.json.JSONException
import org.json.JSONObject


/**
 * This class stores all the information's related to the Errors that can occur.
 */
class NetworkErrorInformation : BaseJSONParser {

    private val STATUS = "statusCode"
    private val KEYERROR = "error"
    private val KEY_MESSAGE = "message"
    private val KEY_DETAILS = "details"
    private val KEY_DETAILS_CODE = "code"
    private val KEY_DETAILS_MESSAGE = "message"

    private var statusCode: Int = 0
    private var error: String? = null
    var message: String? = null
    private var detailCode: Int = 0
    var detailMessage: String? = null

    constructor() {
        //Empty Constructor
    }

    constructor(responseData: ByteArray) {
        val responseString = String(responseData)
        val jsonRegister: JSONObject

        try {
            jsonRegister = JSONObject(responseString)

            statusCode = Integer.parseInt(optInt(jsonRegister, STATUS))
            error = optString(jsonRegister, KEYERROR)
            message = optString(jsonRegister, KEY_MESSAGE)

            val detailObject = optJSONObject(jsonRegister, KEY_DETAILS)
            if (detailObject != null) {

                detailCode = try {
                    Integer.parseInt(optInt(detailObject, KEY_DETAILS_CODE))
                } catch (e: Exception) {
                    e.printStackTrace()
                    0
                }

                detailMessage = try {
                    optString(detailObject, KEY_DETAILS_MESSAGE)
                } catch (e: Exception) {
                    e.printStackTrace()
                    "Unspecified error"
                }

            }

        } catch (e: JSONException) {
            e.printStackTrace()
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun toString(): String {
        return "NetworkErrorInformation{" +
                "statusCode=" + statusCode +
                ", error='" + error + '\''.toString() +
                ", message='" + message + '\''.toString() +
                ", detailCode=" + detailCode +
                ", detailMessage='" + detailMessage + '\''.toString() +
                '}'.toString()
    }
}